cmake/process_users_input.cmake

```diff
@@ -0,0 +1,143 @@
+
+# Library settings:
+option(HDF5 "Use HDF5 library for data output" ON)
+option(NETCDF "Use netcdf library for mesh input" ON)
+option(METIS "Use metis for partitioning" ON)
+option(MPI "Use MPI parallelization" ON)
+option(OPENMP "Use OpenMP parallelization" ON)
+option(ASAGI "Use asagi for material input" OFF)
+
+# todo:
+option(SIONLIB "Use sionlib for checkpointing" OFF)
+option(MEMKIND "Use memkind library for hbw memory support" OFF)
+
+#Seissol specific
+set(ORDER 6 CACHE STRING "Convergence order")  # must be INT type, by cmake-3.16 accepts only STRING
+set(ORDER_OPTIONS 2 3 4 5 6 7 8)
+set_property(CACHE ORDER PROPERTY STRINGS ${ORDER_OPTIONS})
+
+set(NUMBER_OF_MECHANISMS 0 CACHE STRING "Number of mechanisms")
+
+
+set(EQUATIONS "elastic" CACHE STRING "Equation set used")
+set(EQUATIONS_OPTIONS elastic viscoelastic viscoelastic2)
+set_property(CACHE EQUATIONS PROPERTY STRINGS ${EQUATIONS_OPTIONS})
+
+
+set(ARCH "hsw" CACHE STRING "Type of the target architecture")
+set(ARCH_OPTIONS noarch wsm snb hsw knc knl skx)
+set(ARCH_ALIGNMENT   16  16  32  32  64  64  64)  # size of a vector registers in bytes for a given architecture
```

Comment:
I guess before we used a lookup table where names were corresponded with alignment (in bytes). I think it is better to keep this info as close as possible. It will be easy for an extension.  For example, as soon as you decide to add a new cpu architecture you know that you have to manually add its alignment and you will be able to see where you have to add it (just next line)